src/PKSim.Core/Model/OriginData.cs

```diff
@@ -1,58 +1,134 @@
+      };
+
+      public void Deconstruct(out double value, out string unit)
+      {
+         value = Value;
+         unit = Unit;
+      }
+
+      public void Deconstruct(out double value, out string unit, out string name)
+      {
+         value = Value;
+         unit = Unit;
+         name = Name;
+      }
+
+      public OriginDataParameter()
+      {
+      }
+
+      public OriginDataParameter(double value, string unit = "")
+      {
+         Value = value;
+         Unit= unit;
+      }
+   }
+
    public class OriginData : IWithCalculationMethods, IWithValueOrigin
    {
+      private DiseaseState _diseaseState;
+      public readonly List<OriginDataParameter> _allDiseaseStateParameters = new List<OriginDataParameter>();
       public virtual Species Species { get; set; }
       public virtual SpeciesPopulation SpeciesPopulation { get; set; }
       public virtual SubPopulation SubPopulation { get; set; }
       public virtual Gender Gender { get; set; }
       public virtual CalculationMethodCache CalculationMethodCache { get; private set; }
       public ValueOrigin ValueOrigin { get; }
 
-      public virtual double? Age { get; set; }
-      public virtual string AgeUnit { get; set; }
+      public virtual OriginDataParameter Age { get; set; }
+      public virtual OriginDataParameter GestationalAge { get; set; }
+      public virtual OriginDataParameter Height { get; set; }
+      public virtual OriginDataParameter BMI { get; set; }
+      public virtual OriginDataParameter Weight { get; set; }
 
-      public virtual double? GestationalAge { get; set; }
-      public virtual string GestationalAgeUnit { get; set; }
+      // Disease state associated with current origin data. null if no disease state associated with origin data (Healthy)
+      // This is a reference to the db instance and will not be manipulated
+      public virtual DiseaseState DiseaseState
+      {
+         get => _diseaseState;
+         set
+         {
+            _allDiseaseStateParameters.Clear();
+            _diseaseState = value;
+         }
+      }
 
-      public virtual double? Height { get; set; }
-      public virtual string HeightUnit { get; set; }
+      public IReadOnlyList<OriginDataParameter> DiseaseStateParameters => _allDiseaseStateParameters;
 
-      public virtual double? BMI { get; set; }
-      public virtual string BMIUnit { get; set; }
+      public void AddDiseaseStateParameter(OriginDataParameter diseaseStateParameter) => _allDiseaseStateParameters.Add(diseaseStateParameter);
 
-      public virtual double Weight { get; set; }
-      public virtual string WeightUnit { get; set; }
+      public virtual bool IsPreterm
+      {
+         get
+         {
+            var gaValue = GestationalAge?.Value;
+            return gaValue.GetValueOrDefault(CoreConstants.NOT_PRETERM_GESTATIONAL_AGE_IN_WEEKS) <= CoreConstants.NOT_PRETERM_GESTATIONAL_AGE_IN_WEEKS;
+         }
+      }
 
+      public virtual bool IsRealPreterm
```

Comment:
I have added this method with the < but I am not sure why we have this difference. @Yuri05 Can you remember?